### PR TITLE
add support for --allow-use-as-root-and-accept-consequences

### PR DESCRIPTION
**Pull request by boegel**
cfr. #420

### CODE REVIEW

**Comment by boegel** [comment]
@rjeschmi please review?

File: test/framework/options.py [hunk 1]
```diff
@@ -3048,6 +3051,26 @@ def test_check_style(self):
         for pattern in patterns:
             self.assertTrue(re.search(pattern, stdout, re.M), "Pattern '%s' found in: %s" % (pattern, stdout))
 
+    def test_allow_use_as_root(self):
+        """Test --allow-use-as-root"""
+
+        # pretend we're running as root by monkey patching os.getuid used in main
+        easybuild.main.os.getuid = lambda: 0
```
Comment (JensTimmerman) [inline test/framework/options.py]: some linting tools don't like it when you assign lambda's, but this one is so trivial it doesn't require a def imho.

**Comment by JensTimmerman** [comment]
lgtm

**Comment by boegel** [comment]
@JensTimmerman Any concerns about running everything as root vs only the steps that would really require root access (cfr. your comments in #420)?

The latter would be a lot more difficult, and may not be worth the effort, since it wouldn't provide much protection against nasty things from happening during `make install` for example...

**Comment by rjeschmi** [comment]
I think this is fine in general. I think another useful option is to demote from root to a particular user. I have a wrapper that does that for me so I can run easybuild as root more easily.

**Comment by JensTimmerman** [comment]
@boegel I would add -i-know-this-is-a-bad-idea to the end of the option to make it clear this is an option that actually isn't supposed to be used.

**Comment by pescobar** [comment]
hahahaha +1 for @JensTimmerman suggestion ;)

**Comment by boegel** [comment]
@JensTimmerman That's what the warning is for. Making the option annoyingly long/silly is something we can do, but it won't make any difference if you set it via the environment or in a configuration file (which makes it do once and forget).

**Comment by JensTimmerman** [comment]
@boegel sure you can once and forget, but the point is that a new user can immediately identify this option as something that shouldn't be just be used lightheartedly. 
The warning surely helps here, but once the warning is shown it's already to late...

**Comment by boegel** [comment]
OK, fair point @JensTimmerman...

So, pick one:

* `--allow-use-as-root-even-though-it-is-a-bad-idea`
* `--allow-use-as-root-despite-being-not-recommended`
* `--allow-use-as-root-and-accept-consequences`

**Comment by pescobar** [comment]
`--allow-use-as-root-even-though-it-is-a-bad-idea-but-I-am-a-shell-cowboy`

;)

**Comment by JensTimmerman** [comment]
first one

**Comment by boegel** [comment]
We briefly discussed this during today's EasyBuild conf call, where @pescobar mentioned that there may actually be good reasons to install something with root privileges, e.g. installing Singularity with EasyBuild (which requires enabling `setuid` on a binary).

It would be nicer to have support to only leverage admin privileges when really needed, i.e. only when actually enabled `setuid` on something, rather than doing the whole installation as root, but this would require quite a bit more work.

There also seemed to be a consensus that `--allow-use-as-root-and-accept-consequences` is a better option name, since it makes it clear that this configuration setting should be used with care, while not making claims that doing so is *always* wrong (as opposed to `--allow-use-as-root-even-though-it-is-a-bad-idea`). So, I renamed the option to `--allow-use-as-root-and-accept-consequences`.

The idea is to include this support for `--allow-use-as-root-and-accept-consequences` in the upcoming EasyBuild v3.2.0.

**Pull request merged**